docs/scratch/modifications.md

```diff
@@ -0,0 +1,92 @@
+# Scratch Modifications
+
+As part of this project, we modified scratch.
+This file lists and describes the changes.
+
+## Edit and Solve mode
+
+We extended the scratch GUI component to additionally accept a `canEditTask` property which, if disabled, deactivates the following built-in sratch features:
+- the button to load extensions is not shown
+- the buttons to add new sprites is not shown
+- the buttons to load stage sprites is not shown
+- the buttons to delete, duplicate or export sprites is not shown
+- the rendered stage is made non-interactive using `pointer-events: none`
+
+In the following we will use "Task editing mode" (or "Edit mode") to denote `canEditTask=true` and "Task solve mode" (or "Solve mode") to denote `canEditTask=false`.
+
+Moreover, we added the properties `isCostumesTabEnabled` and `isSoundsTabEnabled` which allow hiding the costume and sound tabs.
+
+## CRT Config
+
+To store additional information with a scratch project, we add an additional `crt.json` file into the zip archive.
+
+## Block Config
+
+To restrict the available blocks, we introduced the block config button rendered at the top left of each button in the flyout toolbox to the left.
+The button shows either a number or the infinity symbol, indicting the number of times this block can be used by a student.
+
+In the task editing mode, clicking this button opens a modal allowing the editor to configure whether the block can be used by students and if so, how many times.
+After clicking save in the modal, the updated number is shown on the block.
+
+Note that in editing mode, all blocks are shown in the flyout toolbox, even if they are completely disabled.
+Otherwise it would be impossible for editors to re-enable blocks.
+Moreover, the numbers do not change when blocks are added to targets.
+
+In solve mode, the label shown on the buttons is updated whenever a button is added to a target to keep track of the remaining blocks of that type.
+Moreover, when the counter reaches zero, the block is completely disabled using `pointer-events: none`.
+
+Blocks which the editor configures to be disallowed (a count of `0`), are never rendered in solve mode.
+However, blocks reaching a remaining count of zero are still shown.
+
+### Implementation
+
+Because the blocks part of the scratch GUI is not rendered in react, we need to directly modify the DOM resulting in very non-react code.
```

Comment:
```suggestion
Because the blocks part of the Scratch GUI is not rendered in React, we need to directly modify the DOM, which leads to unorthodox code in a React project.
```